QGL/ChannelLibraries.py

```diff
@@ -657,37 +657,50 @@ def set_master(self, master_instrument, trig_channel=None, pulse_length=1e-7):
         else:
             raise ValueError(f"Could not determine which transmitter to set as master for {master_instrument}:{trig_channel}")
 
+# Used by QGL2, which needs a non-class member function to
+# retrieve a Qubit from the CL without accessing the CL directly
 def QubitFactory(label):
     ''' Return a saved qubit channel'''
+    if channelLib is None:
+        raise Exception("No channel library initialized")
     channelLib.update_channelDict()
     cs = [c for c in channelLib.channelDatabase.channels if c.label==label]
+#    cs = [c for c in channelLib.channelDatabase.channels if c.label==label and isinstance(c, Channels.Qubit)]
```

Comment:
just curious, why don't you want to check that it's a qubit or other channel?

Reply:
I thought it would make sense, but honestly I wasn't sure if there was a reason it hadn't been there before.  It would be better to include that extra check, if it isn't going to hurt something else.

Reply:
Oh okay, I misread, I thought you commented it out. I think it makes sense too

Reply:
OK, added those checks